lib/validate_partitions.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+package validate_partitions;
+use Exporter 'import';
+our @EXPORT = qw(validate_partition_table);
+
+use strict;
+use warnings;
+use scheduler 'get_test_suite_data';
+use testapi;
+use Test::Assert ':all';
+
+
+
+sub validate_partition_table {
+    my ($args) = @_;
+    my $expected_table_type;
+    my $root_disk;
+
+    if (exists($args->{root_disk})) { $root_disk = $args->{root_disk} }
+    else {
+        my @mount_output = (split /\n/, script_output('mount'));
+        my $root_disk;
+        foreach ( @mount_output ) {
+            if ( $_ =~ /(\/dev\/.da).+$/ ) {
+             $root_disk = $1;
+            }
+        }
+    }
+
+    if   (exists($args->{table_type})) { $expected_table_type = $args->{table_type} }
+    else                               { $expected_table_type = (check_var('BACKEND', 's390x')) ? "dasd" : "gpt" }
+
+
+    my @fdisk_output = split(/\n/, script_output("fdisk -l $root_disk"));
+    my $table_type; 
+    foreach ( @fdisk_output ) {
+         if ( $_ =~ /Disklabel\stype: (.+)$/ ) {
+             $table_type = $1;
```

Comment:
you can use capture groups instead of `$1`.

Reply:
From what I found (http://www.wellho.net/resources/ex.php4?item=p212/n2), this is already one way of using capture group. Do you have an example of what you have in mind ?